pinot-spi/src/main/java/org/apache/pinot/spi/ingestion/batch/BatchConfig.java

```diff
@@ -27,41 +27,49 @@
  * Provides all config related to the batch data source, as configured in the table config's ingestion config
  */
 public class BatchConfig {
-  private final Map<String, String> _batchConfigMap;
-  private final String _tableNameWithType;
-
-  private final FileFormat _inputFormat;
-  private final String _inputDirURI;
-  private final String _inputFsClassName;
-  private final Map<String, String> _inputFsProps;
-
-  private final String _outputDirURI;
-  private final String _outputFsClassName;
-  private final Map<String, String> _outputFsProps;
-  private final boolean _overwriteOutput;
-
-  private final String _recordReaderClassName;
-  private final String _recordReaderConfigClassName;
-  private final Map<String, String> _recordReaderProps;
-
-  private final String _segmentNameGeneratorType;
-  private final Map<String, String> _segmentNameGeneratorConfigs;
-  private final String _segmentName;
-  private final String _segmentNamePrefix;
-  private final String _segmentNamePostfix;
-  private final boolean _excludeSequenceId;
-  private final String _sequenceId;
-
-  private final String _pushMode;
-  private final int _pushAttempts;
-  private final int _pushParallelism;
-  private final long _pushIntervalRetryMillis;
-  private final String _pushSegmentURIPrefix;
-  private final String _pushSegmentURISuffix;
-  private final String _pushControllerURI;
-  private final String _outputSegmentDirURI;
+  private Map<String, String> _batchConfigMap;
```

Comment:
Keep them final and change `indexConfigKeys()` to a constructor

Reply:
this is done